### ISSUE TITLE
False negative PossiblyUndefinedIntArrayOffset

### ISSUE THREAD

**Issue by AndrolGenhald** [vimeo/psalm#7125]
I'm not sure how to enable `ensure_array_int_offsets_exist` on psalm.dev, if it's possible, but this is a false negative I found locally: https://psalm.dev/r/2f4e32bae8

**Comment by orklah** [comment]
This is on purpose, `PossiblyUndefinedIntArrayOffset` is only emitted when the offset contain at least one literal int: https://github.com/vimeo/psalm/blob/175ba833326b1283b1b757b7f526616d3fd2383f/src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ArrayFetchAnalyzer.php#L849

Meaning any use of positive-int, int or int range on they own won't trigger this error. I'd not be opposed to introduce a stricter flag if you want to implement that

**Comment by AndrolGenhald** [comment]
> I'd not be opposed to introduce a stricter flag if you want to implement that

Yeah, as it is it doesn't seem that useful for ensuring offset checks. I was thinking about renaming the current behavior to `PossiblyUndefinedLiteralIntArrayOffset` and adding `PossiblyUndefinedIntArrayOffset` as the more generic variant, so that they could be suppressed independently. Does that seem like a good solution, or would you rather change the config option?

**Comment by orklah** [comment]
I'd rather avoid changing behaviour of current configs. Please create new ones instead.

Note that there is also a string variant of this, it may be convenient to add a strict version for both
